### DISCUSSION TITLE
Should we have a different sample app per feature ?

### DISCUSSION THREAD

**Discussion by p1gp1g** [COVESA/covesa-aosp-sdk#4] [General]
When adding push support (#3), the question of the example was raised: should there be a different sample app for each feature, or a general one?

Having a different example for each feature would probably make things clearer, whereas having a general example gives an overview of the SDK.

**Comment by maximilian-galanis** [comment]
I think that each feature in the SDK should have its own minimal example app. This way, single features remain easy to understand, even if the number of features increases in the future.

Are there any other thoughts on this?

**Comment by khvMX** [comment]
We could also have a single app with multiple Activities.

Pro: easy to bump android versions, do gradle updates, reuse code, etc.
Contra: git might become a mess when number of features increases.

Looking at it pragmatically I would suggest sticking with one "Launcher" app which lists all available sample Activities until it's really required to split, and then refactor into different example apps.
